core/src/com/unciv/logic/automation/NextTurnAutomation.kt

```diff
@@ -364,7 +362,7 @@ object NextTurnAutomation {
                     .flatMap { religion -> religion.getBeliefs(beliefType) }.contains(it.value) 
             }
             .map { it.value }
-            .randomOrNull() // ToDo: Better algorithm
+            .maxByOrNull { ChooseBeliefsAutomation.rateBelief(civInfo, it) }
```

Comment:
No fuzzy factor at all left? Won't players notice the boring predictability?

Reply:
Random factor is done when rating the belief. Not perfect, but good enough for now imo

Reply:
And also, the rating is also based on what tiles you own and other factors that change betwee games, so there is always some fuzziness in that